src/libs/auth.ts

```diff
@@ -421,8 +419,8 @@ export const processUser = (user: OidcUser | null, isSignInEvent: boolean): void
       ...state,
       isSignedIn,
       anonymousId: !isSignedIn && state.isSignedIn ? undefined : state.anonymousId,
-      registrationStatus: isSignedIn ? state.registrationStatus : undefined,
-      termsOfService: initializeTermsOfService(isSignedIn, state),
```

Comment:
I'm not sure we can avoid passing `isSignedIn` here.

This is called in the middle of an `authStore.update` which sets the `isSignedIn` field in `authStore`. So I think reading `isSignedIn` from `authStore` inside of `initializeTermsOfService` would yield the value before this update, not the value assigned in this update.

Reply:
An alternative could be to remove the `initializeTermsOfService` function and place its logic here.

Reply:
Yep, that's an error on my part. 

In general, we are going to strip out the `isSignedIn` logic out of this function (in this PR https://github.com/DataBiosphere/terra-ui/pull/4254) since it is only checking if the user object passed in is null or undefined and split that case into the userUnloaded callback function. So with that in mind, I will leave the `initializeTermsOfService` function it its place for now.